src/Kubernetes.Controller/Protocol/ReceiverOptions.cs

```diff
@@ -2,10 +2,17 @@
 // Licensed under the MIT License.
 
 using System;
+using System.Net.Http;
 
 namespace Yarp.Kubernetes.Protocol;
 
 public class ReceiverOptions
 {
     public Uri ControllerUrl { get; set; }
+
+    public TimeSpan ClientTimeout { get; set; } = TimeSpan.FromSeconds(60);
+
+    public HttpMessageHandler ClientHttpHandler { get; set; }
+
+    public HttpClient Client { get; set; } = default!;
```

Comment:
```suggestion
    public HttpMessageInvoker? Client { get; set; }
```

It feels odd to have multiple settings for the same thing. It's also confusing that the timeout would apply if you set the client, but be ignored otherwise (I don't know why `RemoteAuthenticationOptions` was designed this way).

Can we simplify this to accepting the optional client, and change the type to `HttpMessageInvoker` instead of `HttpClient`, as that is what YARP users who do direct forwarding already use today?